src/jit/codegencommon.cpp

```diff
@@ -6221,14 +6222,18 @@ void CodeGen::genReportGenericContextArg(regNumber initReg, bool* pInitRegZeroed
         regSet.verifyRegUsed(reg);
     }
 
-#if CPU_LOAD_STORE_ARCH
+#if defined(_TARGET_ARM64_)
+    genInstrWithConstant(ins_Store(TYP_I_IMPL), EA_PTRSIZE, reg, genFramePointerReg(),
```

Comment:
Mind using `INS_str` instead of `ins_Store(TYP_I_IMPL)` so I don't forget patching it in my ARM load/store PR?

Reply:
I could, but I wonder if that's the right thing to do.

E.g., in this case we have three different lines of `#ifdef'ed` code that is all doing the same thing. Ideally there should be only one line, with no `#ifdefs`. The arm32/arm64 could be merged if we implemented genInstrWithConstant for arm32. Maybe we should implement `emitIns_AR_R` for arm32/arm64. Or maybe codegen shouldn't go directly to the encoder so much. I think we used to generate code via the inst_* functions in instr.cpp more, which abstracted some of the emitter variations.

Bottom-line: if we start replacing platform-independent instruction choice functions here, it will make it harder to refactor / abstract later, possibly.

I think I'm going to leave this as is for now.